ivy/functional/backends/paddle/statistical.py

```diff
@@ -124,19 +124,31 @@ def mean(
         ret = ret.squeeze()
     return ret.astype(ret_dtype)
 
-
 def prod(
     x: paddle.Tensor,
     /,
-    *,
-    axis: Optional[Union[int, Sequence[int]]] = None,
-    dtype: Optional[paddle.dtype] = None,
+    axis: Optional[Union[int, Tuple[int, ...]]] = None,
+    dtype: Optional[str] = None,
     keepdims: bool = False,
     out: Optional[paddle.Tensor] = None,
 ) -> paddle.Tensor:
-    raise IvyNotImplementedException()
-    # TODO:prod causes segmentation fault
-    return paddle.prod(x, axis=axis, keepdim=keepdims, dtype=dtype)
+    assert isinstance(x, paddle.Tensor), f'Expected x to be a paddle.Tensor, but got {type(x)}'
+
+    assert axis is None or isinstance(axis, int) or (isinstance(axis, tuple) and all(isinstance(a, int) for a in axis)), f'Expected axis to be None or int or tuple of ints, but got {type(axis)}'
+
+    if axis is not None:
+        result = paddle.prod(x, axis=axis, keepdim=keepdims)
+    else:
+        result = paddle.prod(paddle.reshape(x, [-1]), keepdim=keepdims)
+
+    if dtype is not None:
+        result = paddle.cast(result, dtype)
```

Comment:
as per function documentation, casting should be before calculation, unless its an unsupported dtype